tests/unit/test_transforms.py

```diff
@@ -1081,3 +1082,41 @@ def test_can_remove_attrs_by_name_from_specific_items(self):
         actual = transforms.RemoveAttributes(self.source, ids=[("1", "val")], attributes=["x"])
 
         compare_datasets(self, expected, actual)
+
+
+class AnnotationReindexTest:
+    @pytest.fixture
+    def fxt_dataset(self, n_labels=3, n_anns=5, n_items=7) -> Dataset:
+        return Dataset.from_iterable(
+            [
+                DatasetItem(
+                    id=f"item_{item_idx}",
+                    annotations=[
+                        Label(label=random.randint(0, n_labels - 1)) for _ in range(n_anns)
+                    ],
+                )
+                for item_idx in range(n_items)
+            ],
+            LabelCategories.from_iterable([f"label_{label_id}" for label_id in range(n_labels)]),
+        )
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    @pytest.mark.parametrize("reindex_each_item", [True, False])
+    def test_annotation_reindex(self, fxt_dataset: Dataset, reindex_each_item: bool):
+        start = random.randint(0, 10)
+        n_anns = sum([len(item.annotations) for item in fxt_dataset])
+
+        # n_anns != n_unique_ids
+        assert n_anns != len(
+            {(item.id, ann.id) for item in fxt_dataset for ann in item.annotations}
+        )
+
+        fxt_dataset.transform(
+            "annotation_reindex",
+            start=start,
+            reindex_each_item=reindex_each_item,
+        )
+        # n_anns == n_unique_ids
+        assert n_anns == len(
+            {(item.id, ann.id) for item in fxt_dataset for ann in item.annotations}
+        )
```

Comment:
This checks whether they have unique ids since we are trying to obtain a set with the `(item.id, ann.id)` pair key. The size of set should be lower than the number of annotations if they don't have unique ids. It can be seen in those lines coming from the dataset before the transformation.
https://github.com/openvinotoolkit/datumaro/blob/af24ed3a1baf9c9b0f79c3905d210e306b23cd80/tests/unit/test_transforms.py#L1109-L1112